packages/react-components/src/atoms/__tests__/Anchor.test.tsx

```diff
@@ -122,4 +122,20 @@ describe('for an internal link with a router', () => {
       expect(spyScrollIntoView).toHaveBeenCalledWith({ behavior: 'smooth' }),
     );
   });
+
+  it('triggers onclick event for internal links', () => {
```

Comment:
shouldn't the test be the opposite of the test above? so does not scroll it into view if it was like e.g. a `?query` link

Reply:
wait, we had to write custom code for this? o.O I thought with normal react router links it would just work

Reply:
No :(

Reply:
https://reactrouter.com/web/guides/scroll-restoration